### PR TITLE
[doc][yba] systemd user for onprem

### PR DESCRIPTION
**Pull request by ddhodge**
systemd user for onprem
DOC-290
DOC-291
DOC-293

@netlify /preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual/#enable-yugabyte-user-processes-to-run-after-logout

### CODE REVIEW

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md [hunk 1]
```diff
@@ -295,149 +293,41 @@ On each node, perform the following as a user with sudo access:
     sudo systemctl status node_exporter
     ```
 
-## Install backup utilities
+## Enable yugabyte user processes to run after logout
 
-YugabyteDB Anywhere supports backing up YugabyteDB to Amazon S3, Azure Storage, Google Cloud Storage, and Network File System (NFS). For more information, see [Configure backup storage](../../../back-up-restore-universes/configure-backup-storage/).
+To enable services to run even when the `yugabyte` user is not logged in, run the following commands:
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:298]: run as yugabyte user

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md [hunk 2]
```diff
@@ -295,149 +293,41 @@ On each node, perform the following as a user with sudo access:
...
+loginctl enable-linger yugabyte
+vi ~/.bashrc
+export XDG_RUNTIME_DIR=/run/user/$(id -u yugabyte)
+```
 
-If you are not using either file, no changes are required.
+{{< note title="Ulimits on RHEL 8" >}}
 
-<!--
+On RHEL 8 systems, additionally, enter the following commands:
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:308]: on RHEL 8 based systems (AlmaLinux 8, RedHat 8, AmazonLinux 2, etc.)

Comment (chidmuthu) [reply docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:308]: And also move this to after the systemd service files definition as a warning/caution

File: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md [hunk 3]
```diff
@@ -91,9 +91,9 @@ For any third-party Cron scheduling tools, you can disable Crontab and add the f
 Disabling Crontab creates alerts after the universe is created, but they can be ignored. You need to ensure Cron jobs are set appropriately for YBA to function as expected.
   {{< /tip >}}
 
-- Verify that Python 3.6 or later is installed.
+- Verify that Python 3.5-3.8 is installed. v3.6 is recommended.
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md:94]: Add step to set runtime config max version to whatever version of python > 3.7 that they are using.

File: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md [hunk 4]
```diff
@@ -91,9 +91,9 @@ For any third-party Cron scheduling tools, you can disable Crontab and add the f
...
 
-    If you are using Python v3.11 or later, install the selinux python package as follows:
+    If you are using Python later than v3.6, install the selinux python package as follows:
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md:96]: They should install https://pypi.org/project/selinux/  for their corresponding python version. For example, the 3.11 command is shown, but an equivalent could be for whatever version of python they are using. If we put it after they symlink python3 the command is pretty simple python3 -m pip install selinux. We can note here that they might also need pip, which they should configure themselves. https://pip.pypa.io/en/stable/installation/ for details.

File: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md [hunk 5]
```diff
@@ -91,9 +91,9 @@ For any third-party Cron scheduling tools, you can disable Crontab and add the f
...
 
     ```sh
     python3.11 -m pip install selinux
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md:99]: change to python3 -m pip install selinux

File: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md (hunk 4 above)
Comment (chidmuthu) [reply docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md:96]: The RedHat bug for this is https://access.redhat.com/solutions/5674911 which is why we need libselinux-python bindings on higher versions of python

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md [hunk 6]
```diff
@@ -62,13 +60,14 @@ Database servers need incoming TCP/IP access enabled to the following ports, for
 
 | Protocol | Port | Description |
 | :------- | :--- | :---------- |
-| TCP | 22 | SSH (for automatic administration) |
+| TCP | 22 | SSH (optional but recommended for troubleshooting) |
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:63]: This isn't optional, it's still required for configure steps to install YB software.

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md [hunk 7]
```diff
@@ -295,149 +298,27 @@ On each node, perform the following as a user with sudo access:
...
-**Sudo is required to set up these services**
-
-If YugabyteDB Anywhere will be using **cron jobs**, make sure the yugabyte user is allowed to run crontab. If you're using the cron.allow file to manage crontab access, add the yugabyte user to this file. If you're using the cron.deny file, remove the yugabyte user from this file.
-
-YugabyteDB Anywhere **systemd services** to perform the monitoring operations mentioned above, then make sure ...
--->
+```sh
+loginctl enable-linger yugabyte
+vi ~/.bashrc
+export XDG_RUNTIME_DIR=/run/user/$(id -u yugabyte)
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:308]: I think we should just say add export XDG_RUNTIME_DIR=/run/user/$(id -u yugabyte) to /home/yugabyte/.bashrc . Don't need to specifically include the vi they can edit however they want.

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md [hunk 8]
```diff
@@ -697,6 +560,20 @@ As an alternative to setting crontab permissions, you can install systemd-specif
     WantedBy=default.target
     ```
 
+### Ulimits on Red Hat Enterprise Linux 8
+
+On Red Hat Enterprise Linux 8-based systems (Red Hat Enterprise Linux 8, Oracle Enterprise Linux 8.x, Amazon Linux 2), additionally, enter the following commands:
+
+```sh
+vi /etc/systemd/system.conf
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:568]: Same here, we can have a ```sh``` part to show the line DefaultLimitNOFILE=1048576 but then just list the two files they line need to be in

File: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md [hunk 9]
```diff
@@ -91,22 +91,26 @@ For any third-party Cron scheduling tools, you can disable Crontab and add the f
...
-    sudo alternatives --display python3
-    python3 -V
+    python3 -m pip install selinux
     ```
 
+    Refer to [Ansible playbook fails with libselinux-python aren't installed on RHEL8](https://access.redhat.com/solutions/5674911) for more information.
+
+    If you are using Python later than v3.7, set the **Max Python Version (exclusive)** Global Configuration option to the python version. Refer to [Manage runtime configuration settings](../../administer-yugabyte-platform/manage-runtime-config/). Note that only a Super Admin user can modify Global configuration settings.
+
 - Enable core dumps and set ulimits, as follows:
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md:115]: I think we should also link to the other doc here, like you added above. At least temporarily that will communicate these both have similar steps. Maybe say something like "if you are not planning on providing YBA with root access too your node you must also follow the manual provisioning steps here"

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md [hunk 10]
```diff
@@ -295,149 +298,27 @@ On each node, perform the following as a user with sudo access:
     sudo systemctl status node_exporter
     ```
 
-## Install backup utilities
-
-YugabyteDB Anywhere supports backing up YugabyteDB to Amazon S3, Azure Storage, Google Cloud Storage, and Network File System (NFS). For more information, see [Configure backup storage](../../../back-up-restore-universes/configure-backup-storage/).
-
-You can install the backup utility for the backup storage you plan to use, as follows:
-
-- NFS: Install rsync, which YugabyteDB Anywhere uses to perform NFS backups installed during one of the previous steps.
```
Comment (chidmuthu) [inline docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:304]: I think rsync is still required even with YBC?

Comment (chidmuthu) [reply docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:304]: Spoke with @Vars-07 and it is not required, so we can safely remove this too. Should we also remove that from our Ansible playbooks too then?

**Review by chidmuthu — APPROVED** [review]

**Review by Vars-07 — APPROVED** [review]
lgtm!

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md (hunk 10 above)
Comment (Vars-07) [reply docs/content/preview/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/on-premises-manual.md:304]: Makes sense. One scenario I can think off is when someone wants to restore the non ybc based backups - then that will be required. Though it's a rare scenario but can exists.
cc: @kv83821-yb for confirmation as well

File: docs/content/preview/yugabyte-platform/configure-yugabyte-platform/supported-os-and-arch.md [hunk 11]
```diff
@@ -24,9 +24,27 @@ YBA supports deploying YugabyteDB on a variety of [operating systems](../../../r
...
 ### Requirements for all OSes
 
-Python v3.6 or later is required. If you're using YBA to provision nodes in public clouds, be sure the custom AMI you plan to use has Python v3.6 or later installed.
+Verify that Python 3.5-3.8 is installed. v3.6 is recommended.
 
-For more information on requirements for nodes for use in on-premises provider configurations, refer to [Prepare nodes for on-premises deployment](../../install-yugabyte-platform/prepare-on-prem-nodes/).
+In case there is more than one Python 3 version installed, ensure that `python3` refers to the right one. For example:
+
+```sh
+sudo alternatives --set python3 /usr/bin/python3.9
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/yugabyte-platform/configure-yugabyte-platform/supported-os-and-arch.md:32]: Shouldn't it be 3.6 (the recommended one) ?

File: docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md [hunk 12]
```diff
@@ -91,22 +91,26 @@ For any third-party Cron scheduling tools, you can disable Crontab and add the f
...
 
-- Verify that Python 3.6 or later is installed.
+- Verify that Python 3.5-3.8 is installed. v3.6 is recommended.
 
-    If you are using Python v3.11 or later, install the selinux python package as follows:
+    In case there is more than one Python 3 version installed, ensure that `python3` refers to the right one. For example:
 
     ```sh
-    python3.11 -m pip install selinux
+    sudo alternatives --set python3 /usr/bin/python3.9
```
Comment (aishwarya24) [inline docs/content/preview/yugabyte-platform/install-yugabyte-platform/prepare-on-prem-nodes.md:99]:
```suggestion
    sudo alternatives --set python3 /usr/bin/python3.6
```

**Review by aishwarya24 — APPROVED** [review]
LGTM with some comments

**Pull request merged**